src/Tasks/AssemblyDependency/AssemblyInformation.cs

```diff
@@ -578,18 +578,19 @@ internal static string GetRuntimeVersion(string path)
 #endif
                 do
                 {
-                    runtimeVersion = new StringBuilder(bufferLength);
+                    runtimeVersion = new char[bufferLength];
                     hresult = NativeMethods.GetFileVersion(path, runtimeVersion, bufferLength, out _);
                     bufferLength *= 2;
-                } while (hresult == NativeMethodsShared.ERROR_INSUFFICIENT_BUFFER);
+                }
+                while (hresult == NativeMethodsShared.ERROR_INSUFFICIENT_BUFFER);
 
                 if (hresult == NativeMethodsShared.S_OK)
                 {
-                    return runtimeVersion.ToString();
+                    return new string(runtimeVersion);
```

Comment:
Must either use the `dwLength` parameter of `GetFileVersion` or find the terminating 0 in the char array. Otherwise the string is oversized.